src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -294,30 +306,32 @@ public static MediaTypeSegmentWithQuality CreateMediaTypeSegmentWithQuality(stri
             var parsedMediaType = new MediaType(mediaType, start, length: null);
             var parser = parsedMediaType._parameterParser;
 
+            // ??? In every other ParseNextParameter() loop, we treat a parameter as valid as long as it comes before
+            // ??? anything invalid in the media type string. Why is this case different? Should the MediaType
+            // ??? constructor validate the parameters, allowing this loop to exit early and increasing consistency?
             double quality = 1.0d;
             MediaTypeParameter parameter;
             while (parser.ParseNextParameter(out parameter))
             {
                 if (parameter.HasName(QualityParameter))
                 {
+                    // ??? This code does not detect the error if a media type contains 2 q parameters. File a bug?
```

Comment:
After thinking about this, I'm ok with how is currently implemented for the following reasons.

1. It works fine for well formed media types.
2. Multiple q parameters are an error and not a common one. (We haven't seen it as far as I know).
3. We want to be relaxed on how we process input in this type, the constraints being:
    * Works as expected for valid media types.
    * It provides a reasonable implementation for invalid ones. (Picks the last q value).

That said, this deserves a comment and a test recording the current behavior.